sklearn/neighbors/classification.py

```diff
@@ -154,18 +155,38 @@ def predict(self, X):
         n_samples = X.shape[0]
         weights = _get_weights(neigh_dist, self.weights)
 
-        y_pred = np.empty((n_samples, n_outputs), dtype=classes_[0].dtype)
-        for k, classes_k in enumerate(classes_):
-            if weights is None:
-                mode, _ = stats.mode(_y[neigh_ind, k], axis=1)
-            else:
-                mode, _ = weighted_mode(_y[neigh_ind, k], weights, axis=1)
+        if self.outputs_2d_ == 'sparse':
+            y_pred_sparse_multilabel = []
 
-            mode = np.asarray(mode.ravel(), dtype=np.intp)
-            y_pred[:, k] = classes_k.take(mode)
+            for k, classes_k in enumerate(classes_):
 
-        if not self.outputs_2d_:
-            y_pred = y_pred.ravel()
+                if weights is None:
+                    mode, _ = stats.mode(_y[neigh_ind, k].toarray(), axis=1)
```

Comment:
could you please confirm what `neigh_ind` and `k` are exactly so that we can debug what's not working?